src/diffusers/models/transformers/dit_transformer2d.py

```diff
@@ -0,0 +1,268 @@
+        num_embeds_ada_norm (int, optional, defaults to 1000):
+            Number of embeddings for AdaLayerNorm, fixed during training and affects the maximum denoising steps during
+            inference.
+        only_cross_attention (bool, optional, defaults to False):
+            Indicates if only cross-attention layers should be used, ignoring self-attention layers.
+        upcast_attention (bool, optional, defaults to False):
+            If true, upcasts the attention mechanism dimensions for potentially improved performance.
+        norm_type (str, optional, defaults to "ada_norm_zero"):
+            Specifies the type of normalization used, can be 'ada_norm_zero'.
+        norm_elementwise_affine (bool, optional, defaults to False):
+            If true, enables element-wise affine parameters in the normalization layers.
+        norm_eps (float, optional, defaults to 1e-5):
+            A small constant added to the denominator in normalization layers to prevent division by zero.
+        use_linear_projection (bool, optional, defaults to False):
+            Deprecated argument. Will be removed in a future version.
+        num_vector_embeds (bool, optional, defaults to False):
+            Deprecated argument. Will be removed in a future version.
+    """
+
+    _supports_gradient_checkpointing = True
+
+    @register_to_config
+    def __init__(
+        self,
+        num_attention_heads: int = 16,
+        attention_head_dim: int = 72,
+        in_channels: Optional[int] = None,
+        out_channels: Optional[int] = None,
+        num_layers: int = 28,
+        dropout: float = 0.0,
+        norm_num_groups: int = 32,
+        cross_attention_dim: Optional[int] = None,
+        attention_bias: bool = False,
+        sample_size: Optional[int] = None,
+        patch_size: Optional[int] = None,
+        activation_fn: str = "gelu-approximate",
+        num_embeds_ada_norm: Optional[int] = 1000,
+        only_cross_attention: bool = False,
+        upcast_attention: bool = False,
+        norm_type: str = "ada_norm_zero",
+        norm_elementwise_affine: bool = False,
+        norm_eps: float = 1e-5,
+        use_linear_projection: str = False,
+        num_vector_embeds: bool = False,
+    ):
+        super().__init__()
+
+        # Validate inputs.
+        deprecation_message = f"`use_linear_projection` is not a valid argument for {self.__class__.__name__}. Please update the configuration of this model to remove this argument. This argument will be removed from the class constructor in a future version."
+        deprecate("use_linear_projection", "1.0.0", deprecation_message)
+        deprecation_message = f"`num_vector_embeds` is not a valid argument for {self.__class__.__name__}. Please update the configuration of this model to remove this argument. This argument will be removed from the class constructor in a future version."
+        deprecate("num_vector_embeds", "1.0.0", deprecation_message)
```

Comment:
As these two arguments aren't used in DiT Transformer anyway. I have kept this so that the configuration stays backwards-compatible.